keras/utils/data_utils.py

```diff
@@ -55,23 +57,93 @@ def chunk_read(response, chunk_size=8192, reporthook=None):
+                        Options are 'auto', 'tar', 'zip', and None.
+                        'tar' includes tar, tar.gz, and tar.bz files.
+                        The default 'auto' is ['tar', 'zip'].
+                        None or an empty list will return no matches found.
+
+    # Return:
+        True if a match was found and an archive extraction was completed,
+        False otherwise.
+    """
+    if archive_format is None:
+        return False
+    if archive_format is 'auto':
+        archive_format = ['tar', 'zip']
+    if isinstance(archive_format, six.string_types):
+        archive_format = [archive_format]
+
+    for archive_type in archive_format:
+        if archive_type is 'tar':
+            open_fn = tarfile.open
+            is_match_fn = tarfile.is_tarfile
+        if archive_type is 'zip':
+            open_fn = zipfile.ZipFile
+            is_match_fn = zipfile.is_zipfile
+
+        if is_match_fn(file_path):
+            with open_fn(file_path) as archive:
+                try:
+                    archive.extractall(path)
+                except (Exception, KeyboardInterrupt) as e:
+                    if os.path.exists(path):
+                        if os.path.isfile(path):
+                            os.remove(path)
+                        else:
+                            shutil.rmtree(path)
+                    raise
+            return True
+    return False
+
+
 def get_file(fname, origin, untar=False,
-             md5_hash=None, cache_subdir='datasets'):
+             md5_hash=None, cache_subdir='datasets',
+             file_hash=None,
+             hash_algorithm='auto',
+             extract=False,
+             archive_format='auto'):
     """Downloads a file from a URL if it not already in the cache.
 
-    Passing the MD5 hash will verify the file after download
-    as well as if it is already present in the cache.
+    Passing the hash will verify the file after download which also
+    means it will skip re-downloading if it is already present in the cache.
+    To compute the hash use `file_hash` or the command line
+    programs shasum and sha256sum.
 
     # Arguments
         fname: name of the file
         origin: original URL of the file
-        untar: boolean, whether the file should be decompressed
-        md5_hash: MD5 hash of the file for verification
-        cache_subdir: directory being used as the cache
+        untar: Deprecated in favor of 'extract'.
+               boolean, whether the file should be decompressed
+        md5_hash: Deprecated in favor of 'file_hash'.
+                  MD5 hash of the file for verification
+        file_hash: The expected hash string of the file after download,
+                   preferably sha256, but md5 is also supported.
+        cache_subdir: Directory where the file is saved,
+                      ~/.keras/ is the default and /tmp/.keras
+                      is a backup default if the first does not work.
```

Comment:
This is not an accurate description. This is the subdirectory inside the Keras directory where to save the file.